spec/requests/tags_request_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'rails_helper'
+
+RSpec.describe TagsController, type: :request do
+
+  let(:user) { valid_user }
+  let(:contact_tag) { Tag.find_by(name: "contract") }
+  before :each do
+    create_tags
+  end
+
+  describe "GET /tags" do
+    it "returns 200 status and tag" do
+      allow(Tag).to receive(:search).and_return([contact_tag])
+      get tags_path("contract"), {format: :json}, authorization_header
+      expect(status).to eq 200
+      expect(response).to match_response_schema("tag/index")
+    end
+  end
+
+  describe "GET /tags/popular" do
+    it "returns 200 status and popular tags" do
+
+      get popular_tags_path, {format: :json}, authorization_header
+
+      expect(parsed_json).to have_key("contract")
+      expect(status).to eq 200
+    end
+  end
+
+  describe "GET /tags/recent" do
+    it "returns 200 status and recent tags" do
```

Comment:
I think the block names should be more explicit. something like `returns status 200 with all tags ordered by the recently created ones`